realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -309,11 +309,24 @@ public final void removeChangeListeners() {
      */
     void notifyChangeListeners() {
         if (listeners != null && !listeners.isEmpty()) {
-            if (row.getTable() == null) return;
+            boolean notify = false;
 
-            long version = row.getTable().version();
-            if (currentTableVersion != version) {
-                currentTableVersion = version;
+            Table table = row.getTable();
+            if (table == null) {
```

Comment:
All invocation scenarios involve an async ops, except `refresh` & local commits that will also trigger the listeners